### ISSUE TITLE
ISSUE-224: Null Pointer Exception on python-as-map iterator without explicit casting

### ISSUE THREAD

**Issue by jjtolton** [clj-python/libpython-clj#224]
```clojure
(require '[libpython-clj2.python :as py :refer [py.. py**] :reload true])
(py/initialize! :python-executable "venv/bin/python")
(require '[libpython-clj2.require :refer [require-python import-python]])
(import-python)
(py/run-simple-string "
class Thing:
    def __call__(this, data):
        return {a: b for a, b in data.items()}

" )
(py.. (__main__/Thing) (__call__  {}  ))
```
Expected:  `{}`
Actual:
```java
   Traceback (most recent call last): File "<string>", line 4, in
   __call__ File "<string>", line 4, in <dictcomp> File
   "/home/jay/.pyenv/versions/3.7.13/lib/python3.7/_collections_abc.py",
   line 743, in __iter__ for key in self._mapping: Exception:
   java.lang.NullPointerException:java.lang.NullPointerException: null
```

Note:
This works 
```clojure
(py.. (__main__/Thing) (__call__  (python/dict {})  )) ;;=> {}  
``` 

See [conversation](https://clojurians.zulipchat.com/#narrow/stream/215609-libpython-clj-dev/topic/empty-map) for more details

**Comment by behrica** [comment]
The simplest test case for the problem is this:

```
(py/py. (libpython-clj2.python.bridge-as-python/map-as-python {}) __iter__)
```

Not sure, if you consider this public API, but it fails with NPE

This is not fixed by PR #225

**Comment by jjtolton** [comment]
Thanks for the test case!  https://github.com/clj-python/libpython-clj/pull/227
